docs/architecture/assessment-runtime/results-collector.md

```diff
@@ -1 +1,34 @@
-# Results Collector
\ No newline at end of file
+# Results Collector
+
+The Results Collector is responsible for collecting all the observations from the Assessment Action Jobs and processing them to generate metrics and other analysis results.
+
+1. The flow begins with the **Job Manager** sending a request to the ****Job Runner**** to run a specific job.
+2. Upon receiving the request, the ****Job Runner**** proceeds to provide the necessary context to the **Assessment Action Job**.
+3. The **Assessment Action Job** is executed and runs the compliance checks against the target component.
+4. The **Assessment Action Job** then passes these observations to the **Results Collector**.
+5. The **Results Collector** takes the observations and processes them internally. This may involve interpreting, analyzing, or organizing the information.
+6. The **Results Collector** continues its internal operations by generating metrics based on the processed observations. These metrics could represent various measurements or analytics related to the assessment.
+7. The **Results Collector** then communicates with the **Event Bus Client**, passing the generated metrics.
```

Comment:
On other diagrams, `Event Bus Client` was called `Runtime Publisher`. We might want to align them as part of this PR as well